src/neuroconv/tools/testing/mock_ttl_signals.py

```diff
@@ -0,0 +1,290 @@
+"""Author: Cody Baker."""
+from typing import Optional, Union
+from pathlib import Path
+
+import numpy as np
+from numpy.typing import DTypeLike
+from nwbinspector.tools import make_minimal_nwbfile
+from nwbinspector.utils import is_module_installed
+from pynwb import NWBHDF5IO, TimeSeries, H5DataIO
+
+from ...utils import IntType, FloatType, ArrayType, FolderPathType
+
+
+def _check_parameter_dtype_consistency(
+    parameter_name: str,
+    parameter_value: Union[IntType, FloatType],
+    generic_dtype: type,  # Literal[np.integer, np.floating]
+):
+    """Helper for `generate_mock_ttl_signal` to assert consistency between parameters and expected trace dtype."""
+    end_format = "an integer" if generic_dtype == np.integer else "a float"
+    assert np.issubdtype(type(parameter_value), generic_dtype), (
+        f"If specifying the '{parameter_name}' manually, please ensure it matches the 'dtype'! "
+        f"Received '{type(parameter_value).__name__}', should be {end_format}."
+    )
+
+
+def generate_mock_ttl_signal(
+    signal_duration: float = 7.0,
+    ttl_times: Optional[ArrayType] = None,
+    ttl_duration: float = 1.0,
+    sampling_frequency_hz: float = 25000.0,
+    dtype: DTypeLike = "int16",
+    baseline_mean: Optional[Union[IntType, FloatType]] = None,
+    signal_mean: Optional[Union[IntType, FloatType]] = None,
+    channel_noise: Optional[Union[IntType, FloatType]] = None,
+    random_seed: Optional[int] = 0,
+) -> np.ndarray:
+    """
+    Generate a synthetic signal of TTL pulses similar to those seen in .nidq.bin files using SpikeGLX.
+
+    Parameters
+    ----------
+    signal_duration: float, optional
+        The number of seconds to simulate.
+        The default is 5.5 seconds.
```

Comment:
```suggestion
    signal_duration: float, optional, default: 7.0
        The number of seconds to simulate.
```

This is proper numpy-docstring style. See reference [here](https://numpydoc.readthedocs.io/en/latest/format.html#parameters). Can you please adjust all of these to show the default in this style?

Reply:
Is it OK to add units onto those default value descriptions? I.e.,

`default: 7.0 seconds` or `default: 25,000 Hz`? I like attaching those so it's super clear what units the argument is in

Reply:
I'd rather stick to the numpy docstring spec